src/Bicep.Core/Registry/Oci/OciArtifactResult.cs

```diff
@@ -25,14 +25,14 @@ public OciArtifactResult(BinaryData manifestBits, string manifestDigest, Immutab
         private readonly BinaryData manifest;
         private readonly string manifestDigest;
         private readonly OciManifest serializedManifest;
-        private readonly ImmutableDictionary<string, BinaryData> layers;
+        private readonly ImmutableList<(string MediaType, BinaryData Data)> layers;
 
         public Stream ToStream() => manifest.ToStream();
 
         public OciManifest Manifest => serializedManifest;
 
         public string ManifestDigest => manifestDigest;
 
-        public IEnumerable<BinaryData> Layers => layers.Values;
+        public IEnumerable<(string MediaType, BinaryData Data)> Layers => layers;
```

Comment:
This is not introduced by this PR, but I think we can just initialize the properties in the constructor and remove the fields.

Reply:
57a9fca7c6072858c33d42ebd8aec05f79906348